### PR TITLE
Fix build error reporting when code coverage is enabled

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

**Comment by Pilchie** [comment]
Paging @jaredpar for `.ps1` review.

File: build/build.ps1 [hunk 1]
```diff
@@ -115,6 +115,9 @@ function Build {
   $useOpenCover = $useCodecov
 
   & $MsbuildExe $ToolsetBuildProj /m /nologo /clp:Summary /nodeReuse:$nodeReuse /warnaserror /v:$verbosity $logCmd /p:Configuration=$configuration /p:SolutionPath=$solution /p:Restore=$restore /p:QuietRestore=true /p:DeployDeps=$deployDeps /p:Build=$build /p:Rebuild=$rebuild /p:Deploy=$deploy /p:Test=$test /p:IntegrationTest=$integrationTest /p:Sign=$sign /p:Pack=$pack /p:UseCodecov=$useCodecov /p:UseOpenCover=$useOpenCover /p:CIBuild=$ci /p:NuGetPackageRoot=$NuGetPackageRoot $properties
```
Comment (jaredpar) [inline build/build.ps1]: There is another `$MsbuildExe` invocation below that seems like it should be similarly protected.

File: build/build.ps1 [hunk 2]
```diff
@@ -115,6 +115,9 @@ function Build {
...
+  if ((-not $?) -or ($lastExitCode -ne 0)) {
+    throw "Aborting after build failure."
+  }
```
Comment (jaredpar) [inline build/build.ps1]: Consider stealing `Exec-Command*` from build-utils.ps1 in the roslyn repo. It has helpers like `Exec-Console` which make all of the error handling automatic.

File: build/build.ps1 (hunk 1 above)
Comment (sharwell) [reply build/build.ps1]: It's protected via the call site, which is how the original call was protected before I added code coverage support.

Comment (jaredpar) [reply build/build.ps1]: The protection for the `$MsbuildExe` invocation on line 124 seems very fragile. True it's the last line in the `Build` function and we immediately exit after `Build` using `$lastExitCode`. But that seems very fragile to change.

**Comment by sharwell** [comment]
@Pilchie @davkean What should I do here? My original goal was to submit the minimum change that I'm confident corrects the problem at hand and generally preserves the coding style already in use by the rest of the build script. Other refactoring and/or style updates are of course possible, but they introduce other risks at the same time.

**Comment by jaredpar** [comment]
My comments here are just comments. I think your change actually made it better as it validates `$?` and `$lastExitCode`. There are corner cases where Powershell will set one but not the other (this is why I moved to explicit Process and validation of error code). Hence the first, and more crucial, MSBuild check is better. 

Overall though IMHO the latter check is flaky as demonstrated by the need for this PR. I would fix this at some point.

**Comment by Pilchie** [comment]
I'm fine with merging this as is for now.

**Pull request merged**